File: Leetcode 0977 Squares of a Sorted Array [easy] [array]/Squares of a Sorted Array [two-pointers].java
class Solution {
    public int[] sortedSquares(int[] A) {
        int[] result = new int[A.length];
        int start = 0;
        int end = A.length - 1;
        int cur = A.length - 1;
        while (start <= end) {
            if (A[start] * A[start] >= A[end] * A[end]) {
                result[cur] = A[start] * A[start];
                start++;
            } else {
                result[cur] = A[end] * A[end];
                end--;
            }
            cur--;
        }
        
        return result;
    }
}

File: Leetcode 0977 Squares of a Sorted Array [easy] [array]/Squares of a Sorted Array [two-pointers].js
/**
 * @param {number[]} A
 * @return {number[]}
 */
var sortedSquares = function(A) {
    let result = [];
    let start = 0;
    let end = A.length - 1;
    let cur = end;
    while (start <= end) {
        if (A[start] * A[start] >= A[end] * A[end]) {
            result[cur--] = A[start] * A[start];
            start++;
        } else {
            result[cur--] = A[end] * A[end];
            end--;
        }
    }
    
    return result;
};